typing/typecore.ml

```diff
@@ -7725,6 +8069,25 @@ let report_error ~loc env = function
         "@[The optional parameter %a \
          cannot have a polymorphic type.@]"
         Style.inline_code l
+  | Cannot_unify_tfunctor_to_tarrow err ->
+      let sub =
+          [Location.msg
+            "@[This function is module-dependent. The dependency is preserved@ \
+               when the function is passed a static module argument %a@ \
+               or %a. Its argument here is not static, so the type-checker@ \
+               tried instead to change the function type to be non-dependent.@]"
+                Style.inline_code "(module M : S)"
+                Style.inline_code "(module M)"] in
+      report_unification_error ~loc ~sub env err
+        ~type_expected_explanation:Format_doc.Doc.empty
```

Comment:
It is better to leave out the optional argument `type_expected_explanation` to make the distinction between 
* having no syntactic context for the expected type
* having a syntactic context for which the best explanation is no explanation.